pkg/networkserver/grpc_deviceregistry.go

```diff
@@ -45,82 +46,206 @@ func (ns *NetworkServer) Set(ctx context.Context, req *ttnpb.SetEndDeviceRequest
 	if err := rights.RequireApplication(ctx, req.Device.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_DEVICES_WRITE); err != nil {
 		return nil, err
 	}
+
+	gets := append(req.FieldMask.Paths[:0:0], req.FieldMask.Paths...)
+	if ttnpb.HasAnyField(req.FieldMask.Paths, "queued_application_downlinks") {
+		gets = append(gets,
+			"mac_state.device_class",
+		)
+	}
+	if ttnpb.HasAnyField(req.FieldMask.Paths, "mac_state.device_class") {
+		gets = append(gets,
+			"mac_state.current_parameters",
+			"mac_state.desired_parameters",
+			"queued_application_downlinks",
+		)
+	}
+
 	var addDownlinkTask bool
-	dev, err := ns.devices.SetByID(ctx, req.Device.EndDeviceIdentifiers.ApplicationIdentifiers, req.Device.EndDeviceIdentifiers.DeviceID, req.FieldMask.Paths, func(dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
-		paths := req.FieldMask.Paths
+	dev, err := ns.devices.SetByID(ctx, req.Device.EndDeviceIdentifiers.ApplicationIdentifiers, req.Device.EndDeviceIdentifiers.DeviceID, gets, func(dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
```

Comment:
Q; why is `SetByID()` not taking `ttnpb.EndDeviceIdentifiers`?

Reply:
We already discussed this before.
Consistency with `GetByEUI`(in NS and JS) and `SetByEUI`(in JS) and being explicit.

Reply:
I don't agree and I don't see the issues you mention.

The `*ByEUI` version is clear and is not related.

I'm suggesting this also to align with AS, which does exactly this. See [here](https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/applicationserver/registry.go#L30-L33).

Reply:
Please create an issue and let's work/discuss this there. It's out of scope of this PR.